crates/ruff_python_semantic/src/analyze/visibility.rs

```diff
@@ -63,9 +63,9 @@ pub fn is_abstract(decorator_list: &[Decorator], semantic: &SemanticModel) -> bo
 /// Returns `true` if a function definition is a `@property`.
 /// `extra_properties` can be used to check additional non-standard
 /// `@property`-like decorators.
-pub fn is_property(
+pub fn is_property<'a>(
     decorator_list: &[Decorator],
-    extra_properties: &[QualifiedName],
+    mut extra_properties: impl Iterator<Item = QualifiedName<'a>>,
```

Comment:
```suggestion
    mut extra_properties: impl IntoIterator<Item = QualifiedName<'a>>,
```

Reply:
I can't get this to work:

```
error[E0507]: cannot move out of `extra_properties`, a captured variable in an `FnMut` closure
  --> crates/ruff_python_semantic/src/analyze/visibility.rs:75:26
   |
71 |     let mut extra_properties = extra_properties.into_iter();
   |         -------------------- captured outer variable
72 |     decorator_list.iter().any(|decorator| {
   |                               ----------- captured by this `FnMut` closure
...
75 |             .is_some_and(|qualified_name| {
   |                          ^^^^^^^^^^^^^^^^ `extra_properties` is moved here
76 |                 let extra_properties = extra_properties.into_iter();
   |                                        ----------------
   |                                        |
   |                                        variable moved due to use in closure
   |                                        move occurs because `extra_properties` has type `<impl IntoIterator<Item = QualifiedName<'a>> as std::iter::IntoIterator>::IntoIter`, which does not implement the `Copy` trait

```

But it occurs to me anyway that this optimisation (to use a lazy iterator rather than eagerly collecting everything as a slice) is incorrect as I currently have it. I'm using the same iterator on each iteration of the outer loop currently. The iterator could have been exhausted after the first iteration of the outer loop, which would lead to incorrect results.